crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2410,7 +2403,19 @@ impl<'db> TypeInferenceBuilder<'db> {
     /// Adds a new diagnostic.
     ///
     /// The diagnostic does not get added if the rule isn't enabled for this file.
-    fn add_diagnostic(&mut self, node: AnyNodeRef, rule: &str, message: std::fmt::Arguments) {
+    pub(super) fn add_diagnostic(
+        &mut self,
+        node: AnyNodeRef,
+        rule: &str,
+        message: std::fmt::Arguments,
+    ) {
+        self.add_diagnostic_string(node, rule, message.to_string());
+    }
+
+    /// Adds a new diagnostic with a string message.
+    ///
+    /// The diagnostic does not get added if the rule isn't enabled for this file.
+    pub(super) fn add_diagnostic_string(&mut self, node: AnyNodeRef, rule: &str, message: String) {
```

Comment:
What's the motivation for this change? I can't find any use of `add_diagnostic_string` outside of `add_diagnostic?

I would prefer not to expose a `_string` function to avoid usage of `format!`.

Reply:
Oops, that was an accidental holdover from a different aborted approach. Removed!